com.unity.netcode.gameobjects/Runtime/Core/NetworkObject.cs

```diff
@@ -1219,5 +1219,21 @@ internal uint HostCheckForGlobalObjectIdHashOverride()
 
             return GlobalObjectIdHash;
         }
+
+        /// <summary>
+        /// Removes a NetworkBehaviour from the ChildNetworkBehaviours list when destroyed
+        /// while the NetworkObject is still spawned.
+        /// </summary>
+        internal void OnNetworkBehaviourDestroyed(NetworkBehaviour networkBehaviour)
+        {
+            if (networkBehaviour.IsSpawned && IsSpawned)
```

Comment:
Same here, why do we care if the NB IsSpawned - isn't it just the NO that matters?